superscore/client.py

```
@@ -15,13 +19,124 @@ class Client:
+        control_layer: Optional[ControlLayer] = None,
+    ) -> None:
+        if backend is None:
+            # set up a temp backend with temp file
+            backend = BACKENDS['test']
+        if control_layer is None:
+            control_layer = ControlLayer()
+
         self.backend = backend
-        self.cl = ControlLayer()
+        self.cl = control_layer
 
     @classmethod
-    def from_config(cls, cfg=None):
-        raise NotImplementedError
+    def from_config(cls, cfg: Path = None):
+        """
+        Create a client from the configuration file specification.
+
+        Configuration file should be of an "ini" format, along the lines of:
+
+        .. code::
+
+            [backend]
+            type = filestore
+            path = ./db/filestore.json
+
+            [control_layer]
+            ca = true
+            pva = true
+
+        The ``backend`` section has one special key ("type"), and the rest of the
+        settings are passed to the appropriate ``_Backend`` as keyword arguments.
+
+        The ``control_layer`` section has a key-value pair for each available shim.
+        The ``ControlLayer`` object will be created with all the valid shims with
+        True values.
+
+        Parameters
+        ----------
+        cfg : Path, optional
+            Path to a configuration file, by default None.  If omitted,
+            :meth:`.find_config` will be used to find one
+
+        Raises
+        ------
+        RuntimeError
+            If a configuration file cannot be found
+        """
+        if not cfg:
+            cfg = cls.find_config()
+        if not os.path.exists(cfg):
+            raise RuntimeError(f"Superscore configuration file not found: {cfg}")
+
+        cfg_parser = configparser.ConfigParser()
+        cfg_file = cfg_parser.read(cfg)
+        logger.debug(f"Loading configuration file at ({cfg_file})")
+
+        # Gather Backend
+        if 'backend' in cfg_parser.sections():
```

Note that, in general, user typos and extra sections in the config file will fail silently as implemented. Perhaps we need to be noisier for missing sections or alert on extra sections (e.g. if I make a `[bakned]` section or a similar typo I won't notice anything is wrong right away)